File: creational/simpleFactory/simpleFactory.js
/**
 * Class presenting a vehicle for rent
 */

class Coffee {
  brew() {}
}

/**
 * Class presenting Cappucino
 * @extends Coffee
 */
class Cappuccino extends Coffee {
  brew() {
    console.log("Brewing cappuccino");
  }
}

/**
 * Class presenting Espresso
 * @extends Coffee
 */
class Espresso extends Coffee {
  brew() {
    console.log("Brewing espresso");
  }
}

/**
 * Class presenting Latte
 * @extends Coffee
 */
class Latte extends Coffee {
  brew() {
    console.log("Brewing latte");
  }
}

/**
 * Class presenting a simple factory
 * @class
 */
class CoffeeFactory {
  /**
   * @param {String} params - type of coffee to return
   * @return {Coffee} Coffee object
   */
  create(coffeeType) {
    switch (coffeeType) {
      case "ESPRESSO":
        return new Espresso();
      case "LATTE":
        return new Latte();
      case "CAPPUCCINO":
        return new Cappuccino();
      default:
        throw new Error("Wrong coffee type: " + coffeeType);
    }
  }
}

// Lets create our factory
coffeeFactory = new CoffeeFactory();
// Create coffee of type espresso and brew it
espresso = coffeeFactory.create("ESPRESSO");
espresso.brew();
// Create coffee of type latte and brew it
latte = coffeeFactory.create("LATTE");
latte.brew();
// Create coffee of type cappuccino and brew it
cappuccino = coffeeFactory.create("CAPPUCCINO");
cappuccino.brew();

File: creational/simpleFactory/simpleFactory.py
class Coffee:
  def brew(self):
    pass

class Cappuccino(Coffee):
  def brew(self):
    print("Brewing cappuccino")

class Espresso(Coffee):
  def brew(self):
    print("Brewing espresso")

class Latte(Coffee):
  def brew(self):
    print("Brewing latte")

class CoffeeFactory:
  def create(self, coffeeType):
    if coffeeType == "ESPRESSO":
      coffee = Espresso()
    elif coffeeType == "LATTE":
      coffee = Latte()
    elif coffeeType == "CAPPUCCINO":
      coffee = Cappuccino()
    else:
      raise Exception("Wrong coffee type: " + coffeeType)
    return coffee

def main():
  # Lets create our factory
  coffeeFactory = CoffeeFactory()
  # Create coffee of type espresso and brew it
  espresso = coffeeFactory.create("ESPRESSO")
  espresso.brew()
  # Create coffee of type latte and brew it
  latte = coffeeFactory.create("LATTE")
  latte.brew()
  # Create coffee of type cappuccino and brew it
  cappuccino = coffeeFactory.create("CAPPUCCINO")
  cappuccino.brew()

if __name__ == "__main__":
    main()